flare/otf.py

```diff
@@ -467,13 +485,55 @@ def train_gp(self):
             hyps_mask=self.gp.hyps_mask,
         )
 
-    def compute_mae(self, gp_frcs, dft_frcs):
-        mae = np.mean(np.abs(gp_frcs - dft_frcs))
-        mac = np.mean(np.abs(dft_frcs))
+    def compute_mae(
+        self,
+        gp_energy,
+        gp_forces,
+        gp_stress,
+        dft_energy,
+        dft_forces,
+        dft_stress,
+    ):
 
         f = logging.getLogger(self.output.basename + "log")
-        f.info(f"mean absolute error: {mae:.4f} eV/A")
-        f.info(f"mean absolute dft component: {mac:.4f} eV/A")
+        f.info("Mean absolute errors & Mean absolute values")
+
+        # compute energy/forces/stress mean absolute error and value
+        if not self.force_only:
+            e_mae = np.mean(np.abs(dft_energy - gp_energy))
+            e_mav = np.mean(np.abs(dft_energy))
+            f.info(f"energy mae: {e_mae:.4f} eV")
+            f.info(f"energy mav: {e_mav:.4f} eV")
+
+            s_mae = np.mean(np.abs(dft_stress - gp_stress))
+            s_mav = np.mean(np.abs(dft_stress))
+            f.info(f"stress mae: {s_mae:.4f} eV/A^3")
+            f.info(f"stress mav: {s_mav:.4f} eV/A^3")
+
+        f_mae = np.mean(np.abs(dft_forces - gp_forces))
+        f_mav = np.mean(np.abs(dft_forces))
+        f.info(f"forces mae: {f_mae:.4f} eV/A")
+        f.info(f"forces mav: {f_mav:.4f} eV/A")
+
+        # compute the per-species MAE
+        unique_species = list(set(self.structure.coded_species))
+        per_species_mae = np.zeros(len(unique_species))
+        per_species_mav = np.zeros(len(unique_species))
+        per_species_num = np.zeros(len(unique_species))
+        for a in range(self.structure.nat):
+            species_ind = unique_species.index(self.structure.coded_species[a])
+            per_species_mae[species_ind] += np.mean(
+                np.abs(dft_forces[a] - gp_forces[a])
+            )
+            per_species_mav[species_ind] += np.mean(np.abs(dft_forces[a]))
+            per_species_num[species_ind] += 1
+        per_species_mae /= per_species_num
+        per_species_mav /= per_species_num
+
+        for s in range(len(unique_species)):
+            curr_species = unique_species[s]
+            f.info(f"type {curr_species} forces mae: {per_species_mae[s]:.4f} eV/A")
+            f.info(f"type {curr_species} forces mav: {per_species_mav[s]:.4f} eV/A")
```

Comment:
Consider including this change in the PR description? Will help to keep track of what the changes are

Reply:
added